yt_dlp/extractor/parler.py

```diff
@@ -87,25 +86,22 @@ class ParlerIE(InfoExtractor):
     def _real_extract(self, url):
         video_id = self._match_id(url)
         data = self._download_json(
-            'https://parler.com/open-api/ParleyDetailEndpoint.php', video_id,
-            data=urlencode_postdata({'uuid': video_id}))['data'][0]
-        primary = data['primary']
+            'https://api.parler.com/v0/public/parleys/' + video_id, video_id)['data']
 
-        embed = self._parse_json(primary.get('V2LINKLONG') or '', video_id, fatal=False)
-        if embed:
-            return self.url_result(embed[0], YoutubeIE)
+        if data.get('link'):
+            return self.url_result(data.get('link'), YoutubeIE)
 
         return {
             'id': video_id,
-            'url': traverse_obj(primary, ('video_data', 'videoSrc')),
-            'thumbnail': traverse_obj(primary, ('video_data', 'thumbnailUrl')),
-            'title': '',
-            'description': strip_or_none(clean_html(primary.get('full_body'))) or None,
-            'timestamp': unified_timestamp(primary.get('date_created')),
-            'uploader': strip_or_none(primary.get('name')),
-            'uploader_id': strip_or_none(primary.get('username')),
-            'uploader_url': format_field(strip_or_none(primary.get('username')), None, 'https://parler.com/%s'),
-            'view_count': int_or_none(primary.get('view_count')),
-            'comment_count': int_or_none(traverse_obj(data, ('engagement', 'commentCount'))),
-            'repost_count': int_or_none(traverse_obj(data, ('engagement', 'echoCount'))),
+            'url': traverse_obj(data, ('video', 'videoSrc')),
+            'thumbnail': traverse_obj(data, ('video', 'thumbnailUrl')),
+            'title': strip_or_none(data.get('title')),
+            'description': strip_or_none(clean_html(data.get('body'))) or None,
+            'timestamp': unified_timestamp(data.get('date_created')),
+            'uploader': strip_or_none(traverse_obj(data, ('user', 'name'))),
+            'uploader_id': strip_or_none(traverse_obj(data, ('user', 'username'))),
+            'uploader_url': format_field(traverse_obj(data, ('user', 'username')), None, 'https://parler.com/%s'),
+            'view_count': int_or_none(data.get('views')),
+            'comment_count': int_or_none(data.get('total_comments')),
+            'repost_count': int_or_none(data.get('echos')),
```

Comment:
This could be done as dict key traversal w/transformations:
```suggestion
            **traverse_obj(data, {
                'url': ('video', 'videoSrc'),
                'thumbnail': ('video', 'thumbnailUrl'),
                'title': ('title', {strip_or_none}),
                'description': ('body', {clean_html}, {lambda x: x if x else None}),
                'timestamp': ('date_created', {unified_timestamp}),
                'uploader': ('user', 'name', {strip_or_none}),
                'uploader_id': ('user', 'username', {str}),
                'uploader_url': ('user', 'username', {lambda x: urljoin('https://parler.com/', x)}),
                'view_count': ('views', {int_or_none}),
                'comment_count': ('total_comments', {int_or_none}),
                'repost_count': ('echos', {int_or_none}),
            }),
```
I think it is OK for `description` to be an empty string if that's what the site gives, so the "lambda of truth" isn't really necessary.
`urljoin` needs to be imported